lib/src/util/vulnerabilitySeverityRatings.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+/**
+ * All Possible Vulnerability (CVSS) Severity Ratings.
+ */
+export const allVulnerabilitySeverityRatings = Object.freeze([
+  'none',
+  'low',
+  'medium',
+  'high',
+  'critical'
+] as const);
+
+type SingleDigit = '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9';
+
+export type VulnerabilityScore = `${SingleDigit}.${SingleDigit}` | '10.0';
+
+/**
+ * The type containing only the possible vulnerability CVSS severity rating values.
+ */
+export type VulnerabilitySeverityRating = (typeof allVulnerabilitySeverityRatings)[number];
+
+/**
+ * Lists the ranges of Vulnerability (CVSS) scores and their corresponding severity rating.
+ */
+export const vulnerabilitySeverityAndScoreRangeList = Object.freeze([
+  { min: 9.0, max: 10.0, severity: 'critical' },
+  { min: 7.0, max: 8.9, severity: 'high' },
+  { min: 4.0, max: 6.9, severity: 'medium' },
+  { min: 0.1, max: 3.9, severity: 'low' },
+  { min: 0.0, max: 0.0, severity: 'none' }
+] as const);
+
+/**
+ * Converts a vulnerability CVSS score from a number to a string.
+ * Throws an error when the score is not within the valid range and is not within one decimal place.
+ */
+export const formatVulnerabilityScore = (scoreNumber: number): VulnerabilityScore => {
+  if (scoreNumber < 0 || scoreNumber > 10) {
+    throw new Error('Vulnerability score must be between 0.0 and 10.0.');
+  }
+  if (!Number.isInteger(scoreNumber * 10)) {
+    throw new Error('Vulnerability score must be a valid number with one decimal place.');
+  }
+  return scoreNumber.toFixed(1) as VulnerabilityScore;
+};
+
+/**
+ * Gets the corresponding CVSS severity rating for a given CVSS score string or number.
+ * Throws an error if the score is not within the valid range or is not a valid number with one decimal place.
+ */
+export const getVulnerabilitySeverityRating = (cvssScore: VulnerabilityScore | number): VulnerabilitySeverityRating => {
+  const score = parseFloat((typeof cvssScore === 'number') ? formatVulnerabilityScore(cvssScore) : cvssScore);
+  const matchedRange = vulnerabilitySeverityAndScoreRangeList.find(range => score >= range.min && score <= range.max);
+  if (!matchedRange) {
+    throw new TypeError('Invalid CVSS score.');
+  }
+  return matchedRange.severity;
+};
+
+/**
+ * (Internal use, not exported to library users)
+ * Maps vulnerability CVSS severity ratings to their corresponding score ranges string representation.
+ */
+export const vulnerabilitySeverityScoreRangeMap = Object.freeze({
```

Comment:
I still want to see this derived from `vulnerabilitySeverityAndScoreRangeList` rather than hardcoded.

Reply:
> I still want to see this derived from `vulnerabilitySeverityAndScoreRangeList` rather than hardcoded.

Updated that here: [0111f22](https://github.com/sonatype/sonatype-react-shared-components/pull/1202/commits/0111f225a38436cdc66e24e07aa98f72bfbee9c9)

Reply:
> ( That accidentally committed ramda import was removed in the next commit )

Speaking of ramda, it kind of seems like you've been going out of your way to avoid using it.  You can use it in the implementation, just not in the example code.